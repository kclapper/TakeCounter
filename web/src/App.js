import React from 'react';

import './index.scss';

import { SettingsProvider } from './components/Settings';
import SettingsMenu from './components/Settings/Menu';

import Details from './components/Details';
import Counter from './components/Counter';

export default function App() {
  return <SettingsProvider>
           <div className="d-flex flex-column text-white" style={{ minHeight: "100vh" }}>
             <div className="d-flex justify-content-between ms-4 me-4 mt-2 mb-0 m-sm-4">
               <Details />
               <SettingsMenu />
             </div>
             <Counter />
           </div>
         </SettingsProvider>
}
