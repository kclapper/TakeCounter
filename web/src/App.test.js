import React from 'react';
import { render } from '@testing-library/react';
import { expect, it } from '@jest/globals';

import App from './App';

it('Matches snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
