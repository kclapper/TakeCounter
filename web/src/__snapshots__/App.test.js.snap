// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-column text-white"
    style="min-height: 100vh;"
  >
    <div
      class="d-flex justify-content-between ms-4 me-4 mt-2 mb-0 m-sm-4"
    >
      <div
        class="d-flex justify-content-start align-items-stretch"
      >
        <div
          class="d-none d-sm-block"
          style="flex-grow: 0.25;"
        />
        <div
          class="h3 border-bottom m-2 ms-0"
          contenteditable="true"
          style="outline: none; min-width: 25vw;"
        />
        <div
          class="d-none d-sm-block"
          style="flex-grow: 2;"
        />
      </div>
      <div>
        <button
          aria-controls="settingsMenu"
          class="btn"
          data-bs-target="#settingsMenu"
          data-bs-toggle="offcanvas"
          type="button"
        >
          <svg
            class="bi"
            fill="white"
            height="24"
            width="24"
          >
            <use
              href="bootstrap-icons.svg#gear-wide-connected"
            />
          </svg>
        </button>
        <div
          aria-labelledby="settingsMenu"
          class="w-100 offcanvas offcanvas-start overflow-y-scroll p-4"
          data-bs-theme="dark"
          id="settingsMenu"
          tabindex="-1"
        >
          <div
            class="offcanvas-header"
          >
            <button
              aria-label="Close"
              class="btn-close"
              data-bs-dismiss="offcanvas"
              type="button"
            />
          </div>
          <div
            class="container"
          >
            <h4
              class="row border-bottom"
            >
              Take Counting
            </h4>
            <div
              class="row justify-content-start"
            >
              <h5
                class="col-6 my-auto"
              >
                Mode
              </h5>
              <div
                class="col-6"
              >
                <select
                  aria-label="Mode"
                  class="form-select"
                >
                  <option
                    value="manual"
                  >
                    Manual
                  </option>
                  <option
                    value="fileWatcher"
                  >
                    File Watcher
                  </option>
                </select>
              </div>
            </div>
            <h4
              class="row border-bottom mt-3"
            >
              Keyboard Shortcuts
            </h4>
            <p>
              To change keyboard shortcuts: click the shortcut box, enter the shortcut you want, then click away from the box.
            </p>
            <div
              class="row justify-content-start"
            >
              <h5
                class="col-6 my-auto"
              >
                Increment Count
              </h5>
              <div
                class="col-6 border rounded p-1"
                contenteditable="true"
                style="outline: none; min-width: 20%;"
              />
            </div>
            <div
              class="row justify-content-start"
            >
              <h5
                class="col-6 my-auto"
              >
                Decrement Count
              </h5>
              <div
                class="col-6 border rounded p-1"
                contenteditable="true"
                style="outline: none; min-width: 20%;"
              />
            </div>
            <div
              class="row justify-content-start"
            >
              <h5
                class="col-6 my-auto"
              >
                Reset Count
              </h5>
              <div
                class="col-6 border rounded p-1"
                contenteditable="true"
                style="outline: none; min-width: 20%;"
              />
            </div>
            <div
              class="row justify-content-start border-top mt-3 pt-2"
            >
              <h4
                class="col-6 my-auto"
              >
                Reset
              </h4>
              <div
                class="col-6 d-flex"
              >
                <div
                  class="flex-grow-1"
                />
                <button
                  class="btn btn-outline-secondary"
                  data-bs-placement="top"
                  data-bs-toggle="tooltip"
                  type="button"
                >
                  Reset Default Settings
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="d-flex flex-column align-items-center justify-content-center"
      style="flex-grow: 1;"
    >
      <div
        style="flex-grow: 0.75;"
      />
      <div
        class="d-flex"
      >
        <h1
          class="display-1 me-4 me-md-5"
        >
          Take
        </h1>
        <div
          class="display-1"
          contenteditable="true"
          style="outline: none; min-width: 40%;"
        />
      </div>
      <div
        class=""
        style="flex-grow: 0;"
      >
        <button
          class="btn btn-primary m-1"
          data-bs-placement="bottom"
          data-bs-title="Shift+_"
          data-bs-toggle="tooltip"
          type="button"
        >
          -
        </button>
        <button
          class="btn btn-primary m-1"
          data-bs-placement="bottom"
          data-bs-title="Shift+R"
          data-bs-toggle="tooltip"
          type="button"
        >
          reset
        </button>
        <button
          class="btn btn-primary m-1"
          data-bs-placement="bottom"
          data-bs-title="Shift+Plus"
          data-bs-toggle="tooltip"
          type="button"
        >
          +
        </button>
      </div>
      <div
        style="flex-grow: 1;"
      />
    </div>
  </div>
</DocumentFragment>
`;
